### ISSUE TITLE
Test der Umgebung für private Repos Lab Gruppen

### ISSUE THREAD

**Issue by mmaechtel** [htwg-bsys/htwg-bsys.github.io#1]
- [ ]  Wie werden die Repos basierend auf ein Template erstellt
- [ ]  Wie funktioniert die Anmeldung der Studenten dafür
- [ ] Wie geben die Studenten Ihre Lösung ab -> Pull Request ?
- [ ] Wie kann während des Semester einfach zusätzliche Dateien in allen Lab Repos der Studenten 'geadded' werden?

**Comment by steveeJ** [comment]
> Wie kann während des Semester einfach zusätzliche Dateien in allen Lab Repos der Studenten 'geadded' werden

Dazu müsste man auf jedem Repo ein clone/pull,add,commit,push ausführen was ich schrecklich finde. Im labworks/gitlab Workflow hatte ich hierfür die *(...)_skeleton* Repos angedacht, worauf alle Kursteilnehmer zugreifen können.

**Comment by mmaechtel** [comment]
Ja, das mit dem Skeleton war mein Fallback, aber wenn das so kompliziert wird, wird der Fallback jetzt einfach der Standard. Ich sortiere mal die Punkte oben neu durch

**Comment by mmaechtel** [comment]
Können wir mit den Tools hier was anfangen als Admin eines Kurses auf github?
[https://github.com/olin-computing/classroom-tools](https://github.com/olin-computing/classroom-tools)

**Comment by steveeJ** [comment]
> Können wir mit den Tools hier was anfangen als Admin eines Kurses auf github?
> https://github.com/olin-computing/classroom-tools

Kann ich nicht beurteilen da ich GitHub Classroom und Jupyter nicht kenne

**Comment by steveeJ** [comment]
> Ja, das mit dem Skeleton war mein Fallback, aber wenn das so kompliziert wird, wird der Fallback jetzt einfach der Standard. Ich sortiere mal die Punkte oben neu durch

Ich denke mal dass GitHub Classroom hier etwas bietet, schließlich geht's ja um die Zuteilung von Aufgaben (assignments). Can we switch back to English so the annoyance of translating all the english terms can be avoided :-D

**Comment by steveeJ** [comment]
>> Ja, das mit dem Skeleton war mein Fallback, aber wenn das so kompliziert wird, wird der Fallback jetzt einfach der Standard. Ich sortiere mal die Punkte oben neu durch
> 
> Ich denke mal dass GitHub Classroom hier etwas bietet, schließlich geht's ja um die Zuteilung von Aufgaben (assignments). Can we switch back to English so the annoyance of translating all the english terms can be avoided :-D
According to this introduction video https://classroom.github.com/#getting-started-video there's "Group Assignments". Is this what you are looking for?

**Comment by mmaechtel** [comment]
Ich habe bei anderen classrooms mal irgendwo gesehen das direkte Einladungen zum forken von reps angeboten wurde.  Allerdings habe ich das noch nicht hinbekommen

**Comment by mmaechtel** [comment]
Das was die Videos da zeigen ist noch nicht so prickelnd

**Comment by mmaechtel** [comment]
OK, inzwischen hab ich das mit dem Classroom mal ausprobiert. Im Prinzip macht das Anmelden über Classroom das gleiche wie der gitlab Teil von labworks. Die Studenten bekommen einen Link zu einem (oder mehreren Repo) und können sich dann dort in eine Gruppe anmelden oder eine neue Gruppe dafür bilden (wie labworks). Allerdings haben die Studenten dann ein privates Repo unter der Organisation htwg-bsys mit Schreibzugriffen per Default. Man kann dann also ohne pull-request arbeiten oder ändert das alles von Hand um, damit die Studenten sich das projekt forken müssen .... Letzteres probiere ich jetzt mal aus.
